File: software_challenge_sdk/src/server_api/sc/protocol/requests/TestModeRequest.java
package sc.protocol.requests;

import com.thoughtworks.xstream.annotations.XStreamAlias;
import com.thoughtworks.xstream.annotations.XStreamAsAttribute;
import sc.protocol.responses.ProtocolMessage;

/**
 * Send by administrative client to enable or disable testMode
 */

@XStreamAlias("testMode")
public class TestModeRequest extends ProtocolMessage  implements ILobbyRequest {


  @XStreamAsAttribute
  public boolean testMode;

  public TestModeRequest(boolean testMode) {
    this.testMode = testMode;
  }
}

File: sdk/src/server-api/sc/protocol/requests/TestModeRequest.kt
package sc.protocol.requests

import com.thoughtworks.xstream.annotations.XStreamAlias
import com.thoughtworks.xstream.annotations.XStreamAsAttribute

/** Request by administrative client to toggle testMode. */
@XStreamAlias("testMode")
class TestModeRequest(
        @XStreamAsAttribute
        val testMode: Boolean
): AdminLobbyRequest
